src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalNucleusApi.kt

```diff
@@ -128,8 +129,23 @@ object CrystalNucleusApi {
         if (itemName.contains(" Powder")) return null
         // Books are not directly added to the loot map, but are checked for later.
         if (itemName.startsWith("§fEnchanted")) {
-            unCheckedBooks += amount
-            return null
+            /**
+             * REGEX-TEST: §fEnchanted Book (Lapidary I)
+             */
+            val bookTypePattern by RepoPattern.pattern(
+                "filter.crystalnucleus.run.enchantedbook",
+                "§fEnchanted Book \\((?<type>\\S*).*\\)"
+            )
+            val bookType = bookTypePattern.matcher(itemName).group("type").lowercase()
```

Comment:
instead of matcher + group, you can use matchMatcher directly (there is a matches() call missing anyway)